packages/core/src/passport-adapter.ts

```diff
@@ -51,13 +51,18 @@ export function passportAuth(config: BlitzPassportConfig) {
       "No Passport strategies found! Please add at least one strategy.",
     )
 
-    const strategy = config.strategies.find((strategy) => strategy.name === req.query.auth[0])
-    assert(strategy, `A passport strategy was not found for: ${req.query.auth[0]}`)
+    const blitzStrategy = config.strategies.find(
+      (strategy) => strategy.strategy.name === req.query.auth[0],
+    )
+    assert(blitzStrategy, `A passport strategy was not found for: ${req.query.auth[0]}`)
+
+    const {strategy, authenticateOptions} = blitzStrategy
 
     passport.use(strategy)
+    const strategyName = strategy.name as string
```

Comment:
Do you need the type assertion here?

Reply:
L75 & L84 `passport.authenticate(...)` can't be called with the `strategy.name` as it is a `string | undefined` (optional).

Happy to pass in `strategy.name as string` to those functions instead if you prefer?